src/server/scripts/EasternKingdoms/zone_goldshire.cpp

```diff
@@ -0,0 +1,263 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "MotionMaster.h"
+#include "ObjectAccessor.h"
+#include "GameEventMgr.h"
+#include "GameTime.h"
+#include "CreatureGroups.h"
+
+enum COF_Paths
+{
+    STORMWIND_PATH  = 80500,
+    GOLDSHIRE_PATH  = 80501,
+    WOODS_PATH      = 80502,
+    HOUSE_PATH      = 80503,
+    LISA_PATH       = 80700
+};
+
+enum COF_Waypoints
+{
+    STORMWIND_WAYPOINT  = 57,
+    GOLDSHIRE_WAYPOINT  = 32,
+    WOODS_WAYPOINT      = 22,
+    HOUSE_WAYPOINT      = 35,
+    LISA_WAYPOINT       = 4
+};
+
+enum COF_Sounds
+{
+    BANSHEE_DEATH           = 1171,
+    BANSHEEPREAGGRO         = 1172,
+    CTHUN_YOU_WILL_DIE      = 8585,
+    CTHUN_DEATH_IS_CLOSE    = 8580,
+    HUMAN_FEMALE_EMOTE_CRY  = 6916,
+    GHOSTDEATH              = 3416
+};
+
+enum COF_Creatures
+{
+    NPC_DANA    = 804,
+    NPC_CAMERON = 805,
+    NPC_JOHN    = 806,
+    NPC_LISA    = 807,
+    NPC_AARON   = 810,
+    NPC_JOSE    = 811
+};
+
+enum COF_Events
+{
+    EVENT_WP_START_GOLDSHIRE    = 1,
+    EVENT_WP_START_WOODS        = 2,
+    EVENT_WP_START_HOUSE        = 3,
+    EVENT_WP_START_LISA         = 4,
+    EVENT_PLAY_SOUNDS           = 5,
+    EVENT_BEGIN_EVENT           = 6
+};
+
+struct npc_cameron : public ScriptedAI
+{
+    npc_cameron(Creature* creature) : ScriptedAI(creature)
+    {
+        _started = false;
+    }
+
+    static uint32 SoundPicker()
+    {
+        return RAND(
+            BANSHEE_DEATH,
+            BANSHEEPREAGGRO,
+            CTHUN_YOU_WILL_DIE,
+            CTHUN_DEATH_IS_CLOSE,
+            HUMAN_FEMALE_EMOTE_CRY,
+            GHOSTDEATH
+        );
+    }
+
+    void MoveTheChildren()
+    {
+        std::vector<Position> MovePosPositions =
+        {
+            { -9373.521f, -67.71767f, 69.201965f, 1.117011f },
+            { -9374.94f, -62.51654f, 69.201965f, 5.201081f },
+            { -9371.013f, -71.20811f, 69.201965f, 1.937315f },
+            { -9368.419f, -66.47543f, 69.201965f, 3.141593f },
+            { -9372.376f, -65.49946f, 69.201965f, 4.206244f },
+            { -9377.477f, -67.8297f, 69.201965f, 0.296706f }
+        };
+
+        Trinity::Containers::RandomShuffle(MovePosPositions);
+
+        // first we break formation because children will need to move on their own now
+        for (auto guid : _childrenGUIDs)
+            if (Creature* child = ObjectAccessor::GetCreature(*me, guid))
+                if (child->GetFormation())
+                    child->GetFormation()->RemoveMember(child);
+
+        // Move each child to an random position
+        for (auto i = 0; i < _childrenGUIDs.size(); ++i)
```

Comment:
```cpp
src/server/scripts/EasternKingdoms/zone_goldshire.cpp:113:28:
 fatal error: comparison of integers of different signs:
 'int' and 'std::vector::size_type' (aka 'unsigned long') [-Wsign-compare]
        for (auto i = 0; i < _childrenGUIDs.size(); ++i)
                         ~ ^ ~~~~~~~~~~~~~~~~~~~~~
```

Reply:
wait...
Cant you merge both for (line 113 and line 107) in one ?